internal/db/bundb/bundb.go

```
@@ -123,16 +125,36 @@ func NewBunDBService(ctx context.Context, c *config.Config) (db.DB, error) {
 		conn = WrapDBConn(bun.NewDB(sqldb, pgdialect.New()))
 	case dbTypeSqlite:
 		// SQLITE
+		var err error
 
 		// Drop anything fancy from DB address
 		c.DBConfig.Address = strings.Split(c.DBConfig.Address, "?")[0]
 		c.DBConfig.Address = strings.TrimPrefix(c.DBConfig.Address, "file:")
 
+		// if we are about to try to create/open a sqlite database file, we should do some checking to ensure
+		// that the place this file is about to be opened even exists and allows us to read and write.
+		// if it doesn't, sqlite will return extremely cryptic errors... so we pre-empt this and warn the user ourselves
+		if c.DBConfig.Address != ":memory:" {
+			absPathForLog, err := filepath.Abs(c.DBConfig.Address)
+			if err != nil {
+				absPathForLog = c.DBConfig.Address
+			}
+			logrus.Infof("attempting to open sqlite database at %s", absPathForLog)
+
+			exists, err := util.FilePathExistsAndIsReadWritable(c.DBConfig.Address)
```

The usual style (in GtS at least) is to check the error first, then check the other value. Also, I think if err isn't nil, we should return an error rather than proceeding, since we know it's going to fail anyway. So I'd replace this chunk with something more like:

```go
exists, err := util.FilePathExistsAndIsReadWritable(c.DBConfig.Address)
if err != nil {
	return nil, fmt.Errorf("could not open sqlite db: %s", err)
}

if !exists {
	return nil, errors.New("blah blah blah")
}
```

---

> we know it's going to fail anyway

This is not true. The FilePathExistsAndIsReadWritable code is just a heuristic guess, its not guaranteed to be 100% accurate.

Now that I think about it, essentially trying to re-implement the operating systems permission logic is  really silly, instead `FilePathExistsAndIsReadWritable` should just try to write a test file and then delete it.  That way if it fails, we know for sure sqlite will fail